packages/oidc_core/lib/src/managers/user_manager_base.dart

```diff
@@ -803,6 +794,10 @@ abstract class OidcUserManagerBase {
 
   @protected
   Future<void> handleTokenExpiring(OidcToken event) async {
+    eventsController.add(
+      OidcTokenExpiringEvent.now(currentToken: event),
+    );
+
     final refreshToken = event.refreshToken;
     if (refreshToken == null) {
```

Comment:
we should also check here if the server supports refresh_token grants
```dart
if (!discoveryDocument.grantTypesSupportedOrDefault
      .contains(OidcConstants_GrantType.refreshToken)) {
  //Server doesn't support refresh_token grant.
  return;
}
```

Reply:
Added